xtask/src/fpga/utils.rs

```diff
@@ -9,6 +9,50 @@ use std::{
 
 use mcu_builder::PROJECT_ROOT;
 
+/// Check that host system has all the tools that the xtask FPGA flows depends on.
+pub fn check_host_dependencies() -> Result<()> {
+    let tools = [
+        (
+            "docker --version",
+            "'docker' not found on PATH. Please install docker.",
+        ),
+        (
+            "rsync --version",
+            "'rsync' not found on PATH. Please install rsync.",
+        ),
+        (
+            "cargo nextest --version",
```

Comment:
I'm not sure how path resolution works with cargo installed programs.  Do we want `cargo nextest` here or `cargo-nextest` or are they equivalent?